arch/arm/cpu/armv8/sysinfo.c

```diff
@@ -0,0 +1,55 @@
+#include <smbios.h>
+// #include <sysinfo.h>
+#include <stdio.h>
+
+void sysinfo_get_cache_info(u8 level, struct smbios_type7 *cache_info) {
+	unsigned long clidr_el1, ccsidr_el1;
```

Comment:
Since this is the hardware we are taking to isn't better to map the u32 values to what the register returns?

Something like:

union {    
    struct {    
     u64 line_size:3
     u64 associativity:10;
     u64 numsets:15;
     u64 unknown:4;
     u64 reserved:32;
    } bits_no_ccidx;
    struct {    
     u64 line_size:3;
     u64 associativity:21;
     u64 reserved1:8;
     u64 numsets:24;
     u64 reserved2:8;
    } bits_ccidx_aarch64;
    struct {    
      u64 line_size:3;
      u64 associativity:21;
      u64 reserved:8;
      u64 unallocated:32;
 }bits_ccidx_aarch32;    
    u64 data;
}

Should cover both a32/64 with and without ccidx